src/sugar3/activity/widgets.py

```diff
@@ -396,3 +396,31 @@ def __init__(self):
         self.paste = PasteButton()
         self.insert(self.paste, -1)
         self.paste.show()
+
+class CollabTextEditor(Gtk.Box):
+    def __init__(self,activity):
+        Gtk.Box.__init__(self)
+        self._collab = CollabWrapper(activity)
+        self._collab.connect('message', self.__message_cb)
+        self._collab.setup()
+        self.textview = Gtk.TextView()
+        self.textview.set_editable(True)
+        self.textview.set_cursor_visible(True)
+        self.textview.set_wrap_mode(Gtk.WrapMode.WORD)
+        self.textbuffer = self.textview.get_buffer()
+        self.textbuffer.connect('changed', self.__text_buffer_edited_cb)
+        self.textbuffer.set_text("TESTING")
+        self.textview.show()
+        self.pack_start(self.textview, expand=True, fill=True, padding=0)
+        self.show()
+
+    def __message_cb(self, collab, buddy, message):
+        action = message.get('action')
+	if action == 'entry_changed':
+	    self.textbuffer.set_text(message.get('new_text'))
+
+    def __text_buffer_edited_cb(self,textbuffer):
+	start_iter = textbuffer.get_start_iter()
+	end_iter = textbuffer.get_end_iter()
+	print textbuffer.get_text(start_iter, end_iter, True)
```

Comment:
On Sun, Dec 13, 2015, 10:38 PM Batchu Venkat Vishal <
notifications@github.com> wrote:

In src
<https://github.com/sugarlabs/sugar-toolkit-gtk3/pull/274#discussion_r47443310>
/
<https://github.com/sugarlabs/sugar-toolkit-gtk3/pull/274#discussion_r47443310>
sugar3
<https://github.com/sugarlabs/sugar-toolkit-gtk3/pull/274#discussion_r47443310>
/activity/
<https://github.com/sugarlabs/sugar-toolkit-gtk3/pull/274#discussion_r47443310>
widgets.py
<https://github.com/sugarlabs/sugar-toolkit-gtk3/pull/274#discussion_r47443310>
:

> + self.textbuffer = self.textview.get_buffer() > +
self.textbuffer.connect('changed', self.__text_buffer_edited_cb) > +
self.textbuffer.set_text("TESTING") > + self.textview.show() > +
self.pack_start(self.textview, expand=True, fill=True, padding=0) > +
self.show() > + > + def __message_cb(self, collab, buddy, message): > +
action = message.get('action') > + if action == 'entry_changed': > +
self.textbuffer.set_text(message.get('new_text')) > + > + def
__text_buffer_edited_cb(self,textbuffer): > + start_iter =
textbuffer.get_start_iter() > + end_iter = textbuffer.get_end_iter() > +
print textbuffer.get_text(start_iter, end_iter, True)

Where do these logs get stored? They are not visible in the Log activity
right?

If you enable the debug log level, they will be stored.  See the
dev-environment docs if you want see how to do that.

—
Reply to this email directly or view it on GitHub
<https://github.com/sugarlabs/sugar-toolkit-gtk3/pull/274/files#r47443310>.<img
src="
https://ci4.googleusercontent.com/proxy/igndzllsiMlbQoNsnja6lSwqc118-lIkRYMzgIB96J27JaHMF8XTqNsje_Q00QNboPxkgBXKGndaV7BTIOJqzUxu4vYY9KE6_qX8-8H2ReDboQ3rIL_DJksykS1H3er6TWA194CHRYG0_-_YP2u3T_Q0_VmBbA=s0-d-e1-ft#https://github.com/notifications/beacon/AFvjmlxjI3bwPTNw2Be0saS9Nrpk_SSaks5pPVBCgaJpZM4G0RrG.gif
">